test/test_regression.py

```diff
@@ -0,0 +1,312 @@
+""" Regression tests for https://github.com/nameko/nameko/issues/428
```

Comment:
I wondered whether to keep these tests in the repo. They were helpful for verifying the fix, and are a good reference, but they're quite slow even when they pass (~45 seconds).

I could pull them out into a gist and reference them here and in the issue?